libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerRecognizer.cs

```diff
@@ -158,7 +158,8 @@ public override async Task<RecognizerResult> RecognizeAsync(DialogContext dialog
             List<Metadata> filters = new List<Metadata>();
             if (IncludeDialogNameInMetadata.GetValue(dialogContext.State))
             {
-                filters.Add(new Metadata() { Name = "dialogName", Value = dialogContext.ActiveDialog.Id });
+                string filterDialogName = dialogContext.ActiveDialog.Id.Replace(".main.dialog", string.Empty);
```

Comment:
This is because after cross-train, the the dialogName of filters is set to schema name, while in QnAMakerRecognizer the dialogName is set to schemaName.main.dialog. We should keep the dialog name consistent.

Reply:
We should talk about this more.  Composer went away from doing .main.dialog so I'm reluctant to build this in.

---
In reply to: [438523515](https://github.com/microsoft/botbuilder-dotnet/pull/4045#discussion_r438523515) [](ancestors = 438523515)